### PR TITLE
subsys: net: Fix sntp clock init Kconfig depends

### LINKED ISSUE #77977: `CONFIG_NET_CONFIG_CLOCK_SNTP_INIT` depends on deprecated `CONFIG_POSIX_CLOCK`
I think this should be changed to `depends on SNTP && CONFIG_POSIX_TIMERS`.

https://github.com/zephyrproject-rtos/zephyr/blob/689d1edee1d57f052b1d4572d67618c0b0e2b8a4/subsys/net/lib/config/Kconfig#L209

### PR DESCRIPTION
**Pull request by lucaato**
Fixes #77977.

### CODE REVIEW

File: subsys/net/lib/config/Kconfig [hunk 1]
```diff
@@ -206,7 +206,7 @@ endif # NET_DHCPV6
```
Comment (jukkar) [inline subsys/net/lib/config/Kconfig:206]: Couple of things changes in the commit message. Please change the subject to
`net: config: Fix sntp clock init Kconfig depends`
Also the commit body needs text, write there about why this change is needed.

Comment (lucaato) [reply subsys/net/lib/config/Kconfig:206]: Applied the required changes

**Review by jukkar — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
